vividus-engine/src/main/java/org/vividus/util/ExamplesTableProcessor.java

```diff
@@ -79,27 +80,26 @@ public static String buildExamplesTable(Collection<String> header, Collection<Li
     {
         String valueSeparator = checkForValueSeparator ? determineValueSeparator(data, properties)
                 : properties.getValueSeparator();
+        properties.getProperties().setProperty(VALUE_SEPARATOR_KEY, valueSeparator);
         StringBuilder examplesTableBuilder = new StringBuilder();
-        String rowSeparator = properties.getRowSeparator();
         if (appendTableProperties)
         {
-            appendTableProperties(examplesTableBuilder, properties, rowSeparator);
+            appendTableProperties(examplesTableBuilder, properties, properties.getRowSeparator());
         }
-        appendRow(examplesTableBuilder, header, properties.getHeaderSeparator());
-        data.forEach(row ->
-        {
-            examplesTableBuilder.append(rowSeparator);
-            appendRow(examplesTableBuilder, row, valueSeparator);
-        });
-        return examplesTableBuilder.toString();
+        String stringExampleTable = ExamplesTableStringBuilder.buildExamplesTableString(properties, List.copyOf(header),
```

Comment:
I think that header and data params are always Lists so maybe it's worth to change API to avoid these copyOf operations

Reply:
There are several places in code where header and data  Set<String> and Set<List<String>> accordingly
Ex.: CsvTableTransformer, DatabaseTableTransformer, DistinctingTableTransformer